bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/SemanticTagRegistry.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.semantics;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.common.registry.Registry;
+
+/**
+ * {@link SemanticTagRegistry} tracks all {@link SemanticTag}s from different {@link SemanticTagProvider}s
+ * and provides access to them.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public interface SemanticTagRegistry extends Registry<SemanticTag, String> {
+
+    /**
+     * Retrieves the class for a given id.
+     *
+     * @param tagId the id of the tag. The id can be fully qualified (e.g. "Location_Room_Bedroom") or a segment, if
+     *            this uniquely identifies the tag
+     *            (e.g. "Bedroom").
+     * @return the class for the id or null, if non exists.
+     */
+    @Nullable
+    Class<? extends Tag> getTagClassById(String tagId);
+
+    /**
+     * Builds the fully qualified id for a semantic tag class.
+     *
+     * @param tag the semantic tag class
+     * @return the fully qualified id
+     */
+    String buildId(Class<? extends Tag> tag);
+
+    /**
+     * Checks if a new tag id is valid.
+     *
+     * To be valid, no tag with this id must already exist in the registry, the tag name extracted from this id
+     * must have a valid syntax, the parent tag extracted from this id must already exists in the registry
+     * and no tag with a same name must already exist in the registry.
+     *
+     * @param is a tag in to check
+     * @return true if the provided id is valid, false if not
+     */
+    boolean isNewIdValid(String id);
```

Comment:
I understand this is necessary in the implementation, but why do we need it as public API?

Reply:
It is used by the REST API to create a new semantic tag. It allows checking that the provided data by the API caller is OK.

Reply:
Maybe I should rename this method into `canBeAdded`.

Reply:
Method renamed.